### PR TITLE
[SUBGRAPH] Workflow for Self hosted subgraph #1548

### LINKED ISSUE #1548: [SUBGRAPH] add support for deploying to our self-hosted subgraph to workflow(s)
See https://github.com/superfluid-finance/protocol-monorepo/blob/dev/.github/workflows/handler.deploy-production-subgraphs.yml

Currently our workflows can deploy to the hosted endpoint (thegraph.com/) and to satsuma.

Task:
Add the capability to deploy to our self-hosted endpoint  
(this is mostly for triggering a re-sync after subgraph schema changes. ok if it requires the subgraph to have been initialized before - added to docker-compose, DB created etc.)

### PR DESCRIPTION
**Pull request by mmd-afegbua**

### CODE REVIEW

**Comment by d10r** [comment]
Do we really need distinct workflow files for hosted service / Satsuma / self hosted?
Seems to me that the process is pretty much the same on all of them.
@0xdavinchee what do you think about generalizing into one workflow instead of adding a 3rd one?

**Comment by 0xdavinchee** [comment]
> Do we really need distinct workflow files for hosted service / Satsuma / self hosted?
> Seems to me that the process is pretty much the same on all of them.
> @0xdavinchee what do you think about generalizing into one workflow instead of adding a 3rd one?

I'm quite indifferent, as long as the generalized workflow is as intuitive, explicit and robust as three existing ones, I'd say go for it!

**Comment by hellwolf** [comment]
# Subgraph Environments (for Protocol-Core)

- Feature Endpoint: matic & goerli 
- Dev Endpoints: ALL
- V1 Endpoints: ALL

## Graph-Hosted Endpoints

- Github Workflow: .github/workflows/handler.deploy-hosted-service-subgraph.yml, params:
  - release_branch: # TODO: more appropriate name should be subgraph environment
  - network: all | ...

## Satsuma-Hosted Endpoints

- Github Workflow: .github/workflows/handler.deploy-satsuma-subgraph.yml, params:
  - release_branch: # assuming V1
  - satsuma_version_label: # FIXME: derive from github ref (if it's tagged, use the tag, if it is not use git commit hash)
  - network: all | ...

## Self-Hosted Endpoints

- Github Workflow: .github/workflows/handler.deploy-self-hosted.yml
  - release_branch:
  - network: all | ...

**Comment by hellwolf** [comment]
https://github.com/superfluid-finance/protocol-monorepo/pull/1588 <-- will merge this first to unbreak those automation contract test cases.

File: packages/subgraph/tasks/deploy-self-hosted-network.sh [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+
+# Set default Subgraph node URL and IPFS API if not provided
+IPFS_API=${IPFS_API:-$DEFAULT_IPFS_API}
+
+# Deploy function
+function deploy_subgraph {
+    local version_label="$1"
+    local network="$2"
+    NODE_URL=${EXTERNAL_NODE_URL:-"https://$network.subgraph.x.superfluid.dev/admin/"}
```
Comment (hellwolf) [inline packages/subgraph/tasks/deploy-self-hosted-network.sh]: use PROVIDER_URL_TEMPLATE, and search how it's used in other script.

Comment (hellwolf) [reply packages/subgraph/tasks/deploy-self-hosted-network.sh]: use SUBGRAPH_URL_TEMPLATE

**Review by hellwolf — APPROVED** [review]
great stuff!

**Comment by d10r** [comment]
still too much reduncancy imo. But lets merge and improve in another PR.

**Pull request merged**